assets/js/extensions/google-analytics/index.ts

```diff
@@ -13,89 +17,217 @@ import {
 	getProductFieldObject,
 	getProductImpressionObject,
 	trackEvent,
+	trackCheckoutStep,
+	trackCheckoutOption,
 } from './utils';
 
-const trackListProducts = ( {
-	products,
-	listName = __( 'Product List', 'woo-gutenberg-products-block' ),
-}: {
-	products: Array< ProductResponseItem >;
-	listName: string;
-} ): void => {
-	trackEvent( 'view_item_list', {
-		event_category: 'engagement',
-		event_label: __( 'Viewing products', 'woo-gutenberg-products-block' ),
-		items: products.map( ( product, index ) => ( {
-			...getProductImpressionObject( product, listName ),
-			list_position: index + 1,
-		} ) ),
-	} );
-};
-
-const trackAddToCart = ( {
-	product,
-	quantity = 1,
-}: {
-	product: ProductResponseItem;
-	quantity: number;
-} ): void => {
-	trackEvent( 'add_to_cart', {
-		event_category: 'ecommerce',
-		event_label: __( 'Add to Cart', 'woo-gutenberg-products-block' ),
-		items: [ getProductFieldObject( product, quantity ) ],
-	} );
-};
-
-const trackRemoveCartItem = ( {
-	product,
-	quantity = 1,
-}: {
-	product: CartResponseItem;
-	quantity: number;
-} ): void => {
-	trackEvent( 'remove_from_cart', {
-		event_category: 'ecommerce',
-		event_label: __( 'Remove Cart Item', 'woo-gutenberg-products-block' ),
-		items: [ getProductFieldObject( product, quantity ) ],
-	} );
-};
+/**
+ * Track customer progress through steps of the checkout. Triggers the event when the step changes.
+ *
+ * 1 - Contact information
+ * 2 - Shipping address
+ * 3 - Billing address
+ * 4 - Shipping options
+ * 5 - Payment options
+ */
+addAction(
+	`${ actionPrefix }-checkout-render-checkout-form`,
+	namespace,
+	trackCheckoutStep( 0 )
+);
+addAction(
+	`${ actionPrefix }-checkout-set-email-address`,
+	namespace,
+	trackCheckoutStep( 1 )
+);
+addAction(
+	`${ actionPrefix }-checkout-set-shipping-address`,
+	namespace,
+	trackCheckoutStep( 2 )
+);
+addAction(
+	`${ actionPrefix }-checkout-set-billing-address`,
+	namespace,
+	trackCheckoutStep( 3 )
+);
+addAction(
+	`${ actionPrefix }-checkout-set-phone-number`,
+	namespace,
+	( { step, ...rest }: { step: string; storeCart: StoreCart } ): void => {
+		trackCheckoutStep( step === 'shipping' ? 2 : 3 )( rest );
+	}
+);
+addAction(
+	`${ actionPrefix }-checkout-set-selected-shipping-rate`,
+	namespace,
+	( { shippingRateId }: { shippingRateId: string } ): void => {
+		trackCheckoutOption( {
+			step: 4,
+			option: __( 'Shipping Method', 'woo-gutenberg-products-block' ),
+			value: shippingRateId,
+		} )();
+	}
+);
+addAction(
+	`${ actionPrefix }-checkout-set-active-payment-method`,
+	namespace,
+	( { paymentMethodSlug }: { paymentMethodSlug: string } ): void => {
+		trackCheckoutOption( {
+			step: 6,
+			option: __( 'Payment Method', 'woo-gutenberg-products-block' ),
+			value: paymentMethodSlug,
+		} )();
+	}
+);
+addAction( `${ actionPrefix }-checkout-submit`, namespace, (): void => {
+	trackEvent( 'add_payment_info' );
```

Comment:
@ralucaStan Purchase is the event fired after payment - it's implemented by the GA extension on the "thanks" page. We shouldn't use that here because some payment methods send you off site to make payment.

Reply:
This info could be added as a comment here to prevent any confusion.

Reply:
Added in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3967/commits/be337d6f0a6310774537eab92e281328c935b807